### DISCUSSION TITLE
quest agent, freeze filesystem

### DISCUSSION THREAD

**Discussion by bdronka** [abbbi/virtnbdbackup#162] [Q&A / Support] [answered]
Hello everyone, I have a few questions about using virtnbdbackup.
1. Do I need to install guest-tool on vm to properly perform a backup while the machine is running?
2. I installed quest-tool on all windows vms and on half of the windows vms I get the message: WARNING fs fs - freeze [main]: internal error: unable to execute QEMU agent command 'guest-fsfreeze-freeze': failed to add C :\ to snapshot set:
Could this affect the backup and restoration?
3. When creating a new copy (full), is all incremental data merged?

**Comment by abbbi** [comment, marked as answer]
hi,

1) you don't need the qemu agent but it is recommended. The Qemu agent within the virtual machine will freeze the Filesystem during the short moment where the checkpoint for the backup is created. This ensures better consistency. 
2) This error is related to Qemu agent on Windows, it seems it is unable to freeze the filesystems as requested (i guess it is attempting to create an vss snapshot). You have to search the web for possible solutions, i don't have much experience with Qemu agent on Windows and what is required to make it work. It can affect restauration: it may be that the filesystem of the restored virtual machine needs recovery during first boot. This depends at which state the virtual machine was and if there were i/o heavy operations during backup. You may check the eventlog of the windows system for possible error causes.
3) The Checkpoints (bitmaps) which were created before will be removed, there is no "merge" happening (after all, the backup mechanism is not snapshot based)
